.github/workflows/gh-release.yml

```diff
@@ -10,9 +10,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-
-      - name: Fetch all history for all tags and branches (necessary for setuptools_scm)
-        run: git fetch --prune --unshallow
+        with:
+          fetch-depth: 0 # Fetch all history for all tags and branches (necessary for setuptools_scm)
```

Comment:
Actually I think you don't need all the commits here, right?
Maybe it was copy-and-pasted here?

Reply:
How about this one? It should work without pulling all commits, right?

Reply:
@lkstrp let's try it out?

Reply:
Think so, let's go. 
But I would also merge `gh-release.yml` with `deploy.yml`.

Reply:
Want to do it in this PR or another one?